src/scripts/InstanceScripts/Wotlk/Raid_IceCrownCitadel.cpp

```diff
@@ -0,0 +1,2077 @@
+        aur->setMaxDuration(duration);
+        aur->setTimeLeft(duration);
+    }
+};
+
+//////////////////////////////////////////////////////////////////////////////////////////
+/// Spell: Bone Storm Damage
+class BoneStormDamage : public SpellScript
+{
+public:
+    SpellScriptEffectDamage doCalculateEffect(Spell* spell, uint8_t effIndex, int32_t* dmg) override
+    {
+        if (effIndex != EFF_INDEX_0 || spell->GetUnitTarget() == nullptr)
+            return SpellScriptEffectDamage::DAMAGE_DEFAULT;
+
+        auto distance = spell->GetUnitTarget()->GetDistance2dSq(spell->getCaster());
+        // If target is closer than 5 yards, do full damage
+        if (distance <= 5.0f)
+            distance = 1.0f;
+
+        *dmg = float2int32(*dmg / distance);
+        return SpellScriptEffectDamage::DAMAGE_FULL_RECALCULATION;
+    }
+};
+
+//////////////////////////////////////////////////////////////////////////////////////////
+/// Spell: Bone Spike Graveyard
+class BoneSpikeGraveyard : public SpellScript
+{
+    public:
+        virtual void onAuraApply(Aura* aur)
+        {
+            aur->removeAuraEffect(EFF_INDEX_1);
+
+            if (Creature* marrowgar = static_cast<Creature*>(aur->GetUnitCaster()))
+            {
+                CreatureAIScript* marrowgarAI = marrowgar->GetScript();
+                if (!marrowgarAI)
+                    return;
+
+                uint8_t boneSpikeCount = uint8_t(aur->GetUnitCaster()->GetMapMgr()->pInstance->m_difficulty & 1 ? 3 : 1);
+                std::list<Unit*> targets;
+
+                targets.clear();
+
+                uint8_t target = 0;
+                for (target; target < boneSpikeCount; ++target)
+                    targets.push_back(GetRandomTargetNotMainTank(marrowgar));
+
+                if (targets.empty())
+                    return;
+
+                uint32 i = 0;
+                for (std::list<Unit*>::const_iterator itr = targets.begin(); itr != targets.end(); ++itr, ++i)
+                {
+                    Unit* target = *itr;
+                    target->castSpell(target, BoneSpikeSummonId[i], true);
+                }
+
+
+                std::vector<uint32_t> emoteVector;
+                emoteVector.clear();
+                emoteVector.push_back(SAY_MARR_BONESPIKE_1);// Stick Around
+                emoteVector.push_back(SAY_MARR_BONESPIKE_2);// The only Escape is Darkness 
+                emoteVector.push_back(SAY_MARR_BONESPIKE_3);// More Bones for the offering
+
+                marrowgarAI->sendRandomDBChatMessage(emoteVector);  
+            }
+        }
+
+        virtual SpellCastResult onCanCast(Spell* spell, uint32_t* parameter1, uint32_t* parameter2)
+        {          
+            if (Creature* marrowgar = static_cast<Creature*>(spell->getUnitCaster()))
+
+                if(GetRandomTargetNotMainTank(marrowgar))
+                    return SpellCastResult::SPELL_CAST_SUCCESS;
+
+            return SpellCastResult::SPELL_FAILED_SUCCESS;
```

Comment:
What do you want to do here? If you want the spell cast to fail, use a cast result like SPELL_FAILED_DONT_REPORT